dashboard/app/models/vocabulary.rb

```diff
@@ -93,6 +94,27 @@ def self.sanitize_key(key)
     end.join.gsub(/_+/, '_')
   end
 
+  # Return a version of the given key which does not conflict
+  # with any existing key for the given CourseVersion. We
+  # achieve this through basic guess-and-check; simply append an
+  # arbitrary incrementable value, and increment it until we
+  # find one that works.
+  #
+  # NOTE that this is not currently used in production; it's
+  # currently only used by the ScriptSeedTest, to deal with
+  # the complex seeding logic used in that test.
+  def self.uniquify_key(key, course_version_id)
+    new_key = key.dup
+    suffix = 'a'
+
+    while Vocabulary.exists?(key: new_key, course_version_id: course_version_id)
+      new_key = "#{key}_#{suffix}"
+      suffix = suffix.next
+    end
+
+    new_key
+  end
```

Comment:
I'm still not clear on why the factory method (updated to use letters instead of numbers as the sequence) does not solve this. perhaps the script_json test fixture contains entries which conflict with the keys generated by the factory? anyway, it is fine with me to keep this method and investigate later.

Reply:
the problem is that the script seed test isn't using the factory method, it's using its own logic to build keys:

https://github.com/code-dot-org/code-dot-org/blob/39e267a8b9faec70cdbdb8be37d14ab333ceeea1/dashboard/test/lib/services/script_seed_test.rb#L782-L785
 
Unfortunately, it's also using a naming scheme for the levels that results in conflicts when those names are turned into keys. I looked into making improvements to the test, but everything I tried just resulted in even more failures, each more opaque than the last.

Reply:
To clarify: the factory method will indeed solve this problem in general, which is why we only need to use this method in the script_seed test, not in the `generate_key` method itself. It's just the script seeding test which needs special accommodation.

Reply:
> I looked into making improvements to the test, but everything I tried just resulted in even more failures, each more opaque than the last.

Sorry to hear the test is being such a pain. If you want to open a work item under Q2 Tech Debt and assign it to me, I can look at it after April launch.

Reply:
already done :) https://codedotorg.atlassian.net/browse/PLAT-836